src/main/kotlin/de/smartsquare/starter/mqtt/MqttAutoConfiguration.kt

```diff
@@ -155,24 +155,26 @@ class MqttAutoConfiguration {
     }
 
     @Bean
+    @ConditionalOnMissingBean
```

Comment:
I did not find a better way here to replace the bean in tests. Maybe you have a better idea?
I need to replace it, to simulate a faulty broker connection. When using a non-existing port for instance, the starter aborts with a `MqttBrokerConnectException` which makes the test fail before any assertions

Reply:
I tried some variants and the following setup in the test works:

```kt
@Test
fun `should be unhealthy on connect`() {
    // Replace MqttConnector to prevent connection on startup for broker to stay in "DISCONNECTED" state
    class MqttTestConnectorConfiguration : MqttConnector() {
        override fun stop(callback: Runnable) = callback.run()
        override fun start() = Unit
        override fun isRunning(): Boolean = false
    }

    runner.withPropertyValues("mqtt.version=5")
        .withBean("mqtt5Connector", MqttTestConnectorConfiguration::class.java)
        .withAllowBeanDefinitionOverriding(true)
        .run { context ->
            val indicator = context.getBean<MqttHealthIndicator>()

            indicator.health().status.`should be equal to`(Status.DOWN)
        }
}
```

With this we can avoid the `@ConditionalOnMissingBean`.

Reply:
Regarding a more customizable design of the connector classes: I think you would want heavy changes like this only in tests. But I can see a user wanting to customize the connect options, which is not currently possible.

I would favor composition over inheritance for that though. We could maybe adopt the `Customizer` pattern for this:

```kt
interface Mqtt5ConnectorCustomizer {
	fun customizeConnectOptions(options: Mqtt5Connect)
}
```

And then call all implementations of that in the connector.

Reply:
Okay I see, I just adjusted the test. For the connection options, lets better wait until this actually emerges as a requirement and do it in another PR to not mix concerns here.